flight/Modules/Logging/logging.c

```diff
@@ -232,107 +219,92 @@ static void loggingTask(void *parameters)
 			}
 		}
 
-
-		// If currently downloading a log, close the file
-		if (loggingData.Operation == LOGGINGSTATS_OPERATION_LOGGING && read_open) {
-			PIOS_STREAMFS_Close(streamfs_id);
-			read_open = false;
-		}
-
-		if (loggingData.Operation == LOGGINGSTATS_OPERATION_LOGGING && !write_open) {
-			if (PIOS_STREAMFS_OpenWrite(streamfs_id) != 0) {
-				loggingData.Operation = LOGGINGSTATS_OPERATION_ERROR;
-			} else {
-				write_open = true;
+		switch (loggingData.Operation) {
```

Comment:
This control structure is pre-existing and maybe should survive this PR, but I'm not a fan of it.  Using a FSM in a task is basically a fancy way to write goto statements around the task function.  I understand it also exists so that a UAVO operation can control logging, but also allows a UAVO operation to send us to wrong states, etc.